lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -698,20 +840,44 @@ instance Arbitrary AddressState where
     shrink = genericShrink
 
 instance Arbitrary Address where
-    arbitrary = Address <$> oneof
-        [ (\bytes -> BS.pack (0x83:bytes)) <$> vectorOf 32 arbitrary
-        , (\bytes -> BS.pack (0x84:bytes)) <$> vectorOf 64 arbitrary
-        , do
-            n <- choose (30, 60)
-            let prefix = BS.pack
-                    [ 130       -- Array(2)
-                    , 216, 24   -- Tag 24
-                    , 88, fromIntegral n -- Bytes(n), n > 23 && n < 256
-                    ]
-            addrPayload <- BS.pack <$> vectorOf n arbitrary
-            let crc = BS.pack [26,1,2,3,4]
-            return (prefix <> addrPayload <> crc)
-        ]
+    arbitrary = (unShowFmt . fst)
+        <$> arbitrary @(ShowFmt Address, Proxy 'Testnet)
+
+instance {-# OVERLAPS #-} Arbitrary (ShowFmt Address, Proxy 'Testnet) where
+    arbitrary = do
+        let proxy = Proxy @'Testnet
+        addr <- ShowFmt <$> frequency
+            [ (10, genAddress 0x83 0x84)
+            , (1, genLegacyAddress (30, 100))
+            ]
+        return (addr, proxy)
+
+instance {-# OVERLAPS #-} Arbitrary (ShowFmt Address, Proxy 'Mainnet) where
+    arbitrary = do
+        let proxy = Proxy @'Mainnet
+        addr <- ShowFmt <$> frequency
+            [ (10, genAddress 0x03 0x04)
```

Comment:
the same comment as above

Reply:
Removed in 566610c8e493fdc4de28078841e74261774be4fe